tree/ntuple/v7/test/rfield_class.cxx

```diff
@@ -14,3 +19,74 @@ TEST(RNTuple, TClass) {
    FileRaii fileGuard("test_ntuple_tclass.root");
    auto ntuple = RNTupleWriter::Recreate(std::move(model), "f", fileGuard.GetPath());
 }
+
+TEST(RTNuple, TObject)
+{
+   FileRaii fileGuard("test_ntuple_tobject.root");
+   {
+      auto model = RNTupleModel::Create();
+      model->MakeField<TObject>("obj");
+      auto writer = RNTupleWriter::Recreate(std::move(model), "ntpl", fileGuard.GetPath());
+      auto entry = writer->GetModel().CreateBareEntry();
+
+      auto heapObj = std::make_unique<TObject>();
+      EXPECT_TRUE(heapObj->TestBit(TObject::kIsOnHeap));
+      EXPECT_TRUE(heapObj->TestBit(TObject::kNotDeleted));
+      heapObj->SetUniqueID(137);
+      entry->BindRawPtr("obj", heapObj.get());
+      writer->Fill(*entry);
+
+      heapObj->~TObject();
+      EXPECT_FALSE(heapObj->TestBit(TObject::kNotDeleted));
+      writer->Fill(*entry);
```

Comment:
Interesting, what are we expecting this to do now that the TObject was deleted?

Reply:
Ok, so what is `writer->Fill` going to fill with? A valid but "empty" TObject? What is the semantic meaning of doing what's happening in these few lines?

Reply:
It may not make much sense, semantically. It's here to test that RNTuple does the same than the `TObject` custom streamer.

Reply:
Consider adding a comment that the code is weird looking on purpose so that it is not 'fixed' by a  later maintainer.